lib/src/cli.rs

```diff
@@ -576,15 +578,46 @@ pub(crate) async fn get_storage() -> Result<crate::store::Storage> {
 }
 
 #[context("Querying root privilege")]
-pub(crate) fn require_root() -> Result<()> {
-    let uid = rustix::process::getuid();
-    if !uid.is_root() {
-        anyhow::bail!("This command requires root privileges");
-    }
-    if !rustix::thread::capability_is_in_bounding_set(rustix::thread::Capability::SystemAdmin)? {
-        anyhow::bail!("This command requires full root privileges (CAP_SYS_ADMIN)");
-    }
+pub(crate) fn require_root(is_container: bool) -> Result<()> {
+    ensure!(
+        rustix::process::getuid().is_root(),
+        match is_container {
+            true =>
+                "The user inside the container from which you are running this command must be root",
+            false => "This command must be executed as the root user",
+        }
+    );
+
+    ensure!(
+        rustix::thread::capability_is_in_bounding_set(rustix::thread::Capability::SystemAdmin)?,
+        match is_container {
+            true => "The container must be executed with the podman --privileged flag",
+            false => "This command requires full root privileges (CAP_SYS_ADMIN)",
+        }
+    );
+
     tracing::trace!("Verified uid 0 with CAP_SYS_ADMIN");
+
+    Ok(())
+}
+
+#[context("Querying root podman")]
+pub(crate) fn require_root_podman() -> Result<()> {
+    let proc_self = Dir::open_ambient_dir("/proc/self", cap_std::ambient_authority())?;
+    let uid_map = proc_self.open("uid_map")?;
+    let uid_map = std::io::BufReader::new(uid_map);
+
+    let map_lines = uid_map.lines().collect::<std::io::Result<Vec<_>>>()?;
+
+    // TODO: I'm unaware of more official channels to check for rootless podman, we should
```

Comment:
We have an existing related check in https://github.com/containers/bootc/blob/af4d43f93c4867dd73a491f85025f90bf5bf9406/lib/src/install.rs#L1014